faiss/utils/distances.cpp

```diff
@@ -129,6 +133,51 @@ void fvec_renorm_L2(size_t d, size_t nx, float* __restrict x) {
 
 namespace {
 
+#ifdef ENABLE_DNNL
+/* Find the nearest neighbors for nx queries in a set of ny vectors using oneDNN/AMX */
+template <class BlockResultHandler,  bool use_sel = false>
+void exhaustive_inner_product_seq_dnnl(
```

Comment:
Let's go a step further and move `exhaustive_inner_product_seq_dnnl` and `exhaustive_inner_product_blas_dnnl` (this should be renamed to `exhaustive_inner_product_dnnl` instead) to `cppcontrib/amx/distances_dnnl.h`?

So that the only DNNL logic remaining in `distances.cpp` is the dispatching mechanism between blas and dnnl in `knn_inner_product_select`